client/mm/libxc/bitget.go

```
@@ -0,0 +1,3133 @@
+	}
+
+	return withdrawResp.OrderId, qty, nil
+}
+
+// ConfirmWithdrawal checks the status of a withdrawal
+func (bg *bitget) ConfirmWithdrawal(ctx context.Context, withdrawalID string, assetID uint32) (uint64, string, error) {
+	assetCfg, err := bitgetAssetCfg(assetID)
+	if err != nil {
+		return 0, "", fmt.Errorf("error getting asset cfg for %d: %w", assetID, err)
+	}
+
+	params := map[string]string{
+		"coin": assetCfg.coin,
+	}
+
+	// Add chain parameter for tokens (required by Bitget API)
+	if assetCfg.chain != "" {
+		params["chain"] = assetCfg.chain
+	}
+
+	// Add startTime to limit results (24 hours ago in milliseconds)
+	startTime := time.Now().Add(-24 * time.Hour).UnixMilli()
+	params["startTime"] = strconv.FormatInt(startTime, 10)
+
+	// Add endTime (now in milliseconds)
+	endTime := time.Now().UnixMilli()
+	params["endTime"] = strconv.FormatInt(endTime, 10)
+
+	var resp bgtypes.APIResponse
+	err = bg.getAPI(ctx, "/api/v2/spot/wallet/withdrawal-records", params, true, &resp)
+	if err != nil {
+		return 0, "", err
+	}
+
+	if resp.Code != "00000" {
+		return 0, "", fmt.Errorf("bitget API error: %s - %s", resp.Code, resp.Msg)
+	}
+
+	var withdrawals []*bgtypes.WithdrawalRecord
+	if err := json.Unmarshal(resp.Data, &withdrawals); err != nil {
+		return 0, "", fmt.Errorf("error unmarshaling withdrawal records: %w", err)
+	}
+
+	bg.log.Debugf("ConfirmWithdrawal for ID %s: got %d withdrawal records", withdrawalID, len(withdrawals))
+
+	for _, status := range withdrawals {
+		if status.OrderId == withdrawalID {
+			bg.log.Infof("Found matching withdrawal for ID %s: status=%s, txID=%s", withdrawalID, status.Status, status.TxId)
+
+			// Bitget withdrawal statuses: success, pending, failed, etc.
+			switch status.Status {
+			case "success":
+				if status.TxId == "" {
+					bg.log.Debugf("Withdrawal pending (no txID yet): ID %s", withdrawalID)
+					return 0, "", ErrWithdrawalPending
+				}
+
+				sizeFloat, _ := strconv.ParseFloat(status.Size, 64)
```

Rather not ignore error.